### PR TITLE
Refactor a Debugger notebook and rename as a tutorial

### PR DESCRIPTION
**Pull request by mchoi8739**
*Issue #, if available:*

https://github.com/aws/amazon-sagemaker-examples/issues/1286

*Description of changes:*

* Refactor the original notebook [sagemaker-debugger/tensorflow_builtin_rule/tf-mnist-builtin-rule.ipynb](https://github.com/aws/amazon-sagemaker-examples/blob/master/sagemaker-debugger/tensorflow_builtin_rule/tf-mnist-builtin-rule.ipynb) and rename as a tutorial
* Simplify and update the documentation
* Focus on showing how to use the Debugger built-in rules and smdebug trial
* Improve the final visualization script for plotting accuracy and loss curves

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: sagemaker-debugger/amazon_sagemaker_debugger_tutorial_notebooks/debugger_tutorial_built_in_rules_debugging.ipynb [hunk 1]
```diff
@@ -0,0 +1,535 @@
+{
```
Comment (NihalHarish) [inline sagemaker-debugger/amazon_sagemaker_debugger_tutorial_notebooks/debugger_tutorial_built_in_rules_debugging.ipynb:1]: <p>How do we know that the tensors become available in the S3 buckets in 10 minutes?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/1676/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

**Comment by NihalHarish** [comment]
I see that we have also deleted the byoc training script, do we provide instructions for this elsewhere?

File: sagemaker-debugger/amazon_sagemaker_debugger_tutorial_notebooks/debugger_tutorial_built_in_rules_debugging.ipynb (hunk 1 above)
Comment (mchoi8739) [reply sagemaker-debugger/amazon_sagemaker_debugger_tutorial_notebooks/debugger_tutorial_built_in_rules_debugging.ipynb:1]: I'll add some more description for this. It takes ~5 mins for initiating an instance and another few mins for the training job to start retrieving tensors, save to the S3 bucket, and retrieve tensors. I suggested 10 minutes to give enough time to see tensors

**Comment by mchoi8739** [comment]
> I see that we have also deleted the byoc training script, do we provide instructions for this elsewhere?

I'd rather keep the notebooks to have one topic at a time rather than having everything in each notebook. We have a separate BYOC notebook, so in this notebook we do not need to have byoc script.

File: sagemaker-debugger/amazon_sagemaker_debugger_tutorial_notebooks/debugger_tutorial_built_in_rules_debugging.ipynb (hunk 1 above)
Comment (mchoi8739) [reply sagemaker-debugger/amazon_sagemaker_debugger_tutorial_notebooks/debugger_tutorial_built_in_rules_debugging.ipynb:1]: Incorporated

**Comment by mchoi8739** [comment]
> I see that we have also deleted the byoc training script, do we provide instructions for this elsewhere?

Also the training script is c/p of one of the examples here: https://github.com/awslabs/sagemaker-debugger/blob/master/examples/tensorflow/sagemaker_byoc/mnist.py

File: sagemaker-debugger/amazon_sagemaker_debugger_tutorial_notebooks/debugger_tutorial_built_in_rules_debugging.ipynb (hunk 1 above)
Comment (NihalHarish) [reply sagemaker-debugger/amazon_sagemaker_debugger_tutorial_notebooks/debugger_tutorial_built_in_rules_debugging.ipynb:1]: I feel that this statement might give the impression that debugger takes ~ 10 minutes, but in this particular case, it is the actual sagemaker training job including (container init) that is contributing to the time duration.

Comment (mchoi8739) [reply sagemaker-debugger/amazon_sagemaker_debugger_tutorial_notebooks/debugger_tutorial_built_in_rules_debugging.ipynb:1]: I simply removed "10 minutes". Now the sentence is:

> 
To access the saved tensors by Debugger, use the `smdebug` client library to create a Debugger trial object. The following code cell sets up a `tutorial_trial` object, and waits until it finds available tensors from the default S3 bucket.

I think this statement is actually enough, since the trial returns automatic messages until it finds available tensor.
